sdk-workflows/src/main/java/io/dapr/workflows/runtime/WorkflowRuntime.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2023 The Dapr Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package io.dapr.workflows.runtime;
+
+import com.google.protobuf.Empty;
+import com.microsoft.durabletask.DurableTaskGrpcWorker;
+import com.microsoft.durabletask.DurableTaskGrpcWorkerBuilder;
+import io.dapr.utils.NetworkUtils;
+import reactor.core.publisher.Mono;
+
+/**
+ * Contains methods to register workflows and activities.
+ */
+public class WorkflowRuntime implements AutoCloseable {
+
+  private static volatile WorkflowRuntime instance;
+  private DurableTaskGrpcWorkerBuilder builder;
```

Comment:
The code for this class is a bit confusing, because there are two responsibilities in this class: 

1. how to build up WorkflowRuntime: and WorkflowRuntime should only have one instance
2. workflowRuntime implementation

I suggest to move all the code about how to build up a WorkflowRuntime instance to a standalone class named WorkflowRuntimeBuilder, so that we can get simple and clean code for WorkflowRuntime.

WorkflowRuntimeBuilder class should contain:

- private DurableTaskGrpcWorkerBuilder builder;
- public <T extends Workflow> void registerWorkflow(Class<T> clazz) {}
- a new Build() method to do something like this.worker = this.builder.build();
- the logic that we should only have one instance of WorkflowRuntime